src/reverse.ts

```diff
@@ -1,11 +1,18 @@
 import { purry } from './purry';
 
 // prettier-ignore
-type ReverseTuple<T extends readonly unknown[], R extends readonly unknown[] = []> = ReturnType<
-  T extends readonly [infer F, ...infer L] ? () => ReverseTuple<L, [F, ...R]> : () => R
+type Reverse<
+  T extends readonly unknown[],
+  R extends readonly unknown[] = []
+> = ReturnType<
```

Comment:
@TkDodo Hey I was looking at how Reverse was implemented in remeda and [was told to ping you](https://github.com/remeda/remeda/issues/894#issuecomment-2422846205) about the usage of ReturnType here.

Mind explaining ?  To be clear, I'd expect the type to be written as

```ts
type Reverse<
  T extends readonly unknown[],
  R extends readonly unknown[] = []
> =
  T extends IsNoTuple<T>
    ? [...T, ...R]
    : T extends readonly [infer F, ...infer L]
      ? Reverse<L, [F, ...R]>
      : R
```

I figure there's some typescript behavior being manipulated and was hoping you could explain what.

Reply:
Haha thanks so much for taking the time to respond.  Understandable to not remember !